src/main/java/com/auth0/net/CreateUserRequest.java

```diff
@@ -0,0 +1,21 @@
+package com.auth0.net;
+
+import com.auth0.json.auth.UserInfo;
+import com.fasterxml.jackson.core.type.TypeReference;
+import okhttp3.OkHttpClient;
+
+import java.util.Map;
+
+public class CreateUserRequest extends CustomRequest<UserInfo> implements SignUpRequest {
```

Comment:
Can't this be stand alone and the old Signup is simply deprecated?

Reply:
I don't follow. The idea is that every API method returns an interface that just exposes the methods that the endpoint allows. A `SignUpRequest` would expose a custom fields setter method, for example.